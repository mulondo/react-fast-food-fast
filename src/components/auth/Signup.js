import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {signUp} from '../../actions/signUpAction';

// import food from '../../images/food.jpg';

export class Register extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email:'',
      username:'',
      telephone:'',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e){
    return this.setState({ [e.target.name]: e.target.value });
  }

  
  onSubmit(e) {
    e.preventDefault();
    const newUser = {
      username: this.state.username,
      email: this.state.email,
      password: this.state.password,
      telephone:this.state.telephone
    };
    this.props.signUp(newUser);
    
  }

  render() {
    let userInformation = (
      <div className="signupmodal">
          <form className="modal-content animate" name="sigin" onSubmit={this.onSubmit}>
              <h1>Fast-Food-Fast</h1>
              <div className="imgcontainer">    
                  {/* <img src={food} alt="User" className="food"/> */}
              </div>
              <div className="container">

                    <label forhtml="username"><b>Username</b></label>
                    <input type="text"
                      placeholder="Enter Username" 
                      value={this.state.username} 
                      onChange={this.onChange} 
                      name="username" 
                      required/>

                    <label forhtml="Email"><b>Email</b></label>
                    <input 
                      type="email" 
                      placeholder="example@gmail.com" 
                      name="email" 
                      value={this.state.email}
                      onChange={this.onChange} 
                      required/>

                    <label forhtml="telephone"><b>Phone number</b></label>
                    <input 
                      type="text" 
                      id="telephone" 
                      placeholder="+256-703-492035" 
                      pattern="\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                      value={this.state.telephone}
                      onChange={this.onChange} 
                      name="telephone" 
                      required/>

                    <label forhtml="password"><b>Password</b></label>
                    <input 
                      type="password" 
                      id="pwd"
                      value={this.state.password}
                      onChange={this.onChange}
                      placeholder="***************"
                      name="password" 
                      required/>

                      <input type="submit" className="signupbtn" value="Register"/>
              </div>
        </form>
    </div>
      
    );
    return <div className="flex-grow-1">{userInformation}</div>;
  }
}

// Register.prototype={
//   signUp:PropTypes.func.isRequired
// }

export default connect(null, {signUp})(Register);