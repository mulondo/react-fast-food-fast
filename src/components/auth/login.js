import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import {Login} from '../../actions/login';

// import food from '../../images/food.jpg';

export class Userlogin extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username:'',
      password: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e){
    return this.setState({ [e.target.name]: e.target.value });
  }
  
  onSubmit(e) {
    e.preventDefault();
    const newUser = {
      username: this.state.username,
      password: this.state.password
    };
    this.props.Login(newUser, this.props.history);  
  }

  render() {
    let userInformation = (
      <div className="signupmodal">
          <form className="modal-content animate" name="sigin" onSubmit={this.onSubmit}>
              <h1>Fast-Food-Fast</h1>
              <div className="imgcontainer">    
                  {/* <img src={food} alt="User" className="food"/> */}
              </div>
              <div className="container">
                    <label forhtml="username"><b>Username</b></label>
                    <input type="text"
                      placeholder="Enter Username" 
                      value={this.state.username} 
                      onChange={this.onChange} 
                      name="username" 
                      required/>

                    <label forhtml="password"><b>Password</b></label>
                    <input 
                      type="password" 
                      id="pwd"
                      value={this.state.password}
                      onChange={this.onChange}
                      placeholder="***************"
                      name="password" 
                      required/>
                      <input type="submit" className="signupbtn" value="Login"/>
                      <Link to="/signup">
                      <h4>Sign up</h4>
                      </Link>
              </div>
        </form>
    </div>
      
    );
    return <div className="flex-grow-1">{userInformation}</div>;
  }
}


export default connect(null, {Login})(Userlogin);