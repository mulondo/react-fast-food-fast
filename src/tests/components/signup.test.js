import React from 'react';
import { shallow, mount } from 'enzyme';
import { Register } from '../../components/auth/Signup';


describe('<Register/>', () => {
  const editor = shallow(<Register registerUser={jest.fn()} />);
  const props = {
    registerUser: (user) => jest.fn(),
    signUp: jest.fn()
  };

  it('should render without crashing', () => {
    expect(editor).toMatchSnapshot();
  });

  it('test component on submit', () => {
    const wrapper = mount(<Register {...props} />);
    const instance = wrapper.instance();
    instance.onChange({ target: { name: 'name', value: 'value' } });
    const e = { preventDefault: () => {}};
    instance.onSubmit(e);
  });

});