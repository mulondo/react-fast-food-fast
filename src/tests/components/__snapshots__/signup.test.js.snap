// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    registerUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="signupmodal"
      >
        <form
          className="modal-content animate"
          name="sigin"
          onSubmit={[Function]}
        >
          <h1>
            Fast-Food-Fast
          </h1>
          <div
            className="imgcontainer"
          />
          <div
            className="container"
          >
            <label
              forhtml="username"
            >
              <b>
                Username
              </b>
            </label>
            <input
              name="username"
              onChange={[Function]}
              placeholder="Enter Username"
              required={true}
              type="text"
              value=""
            />
            <label
              forhtml="Email"
            >
              <b>
                Email
              </b>
            </label>
            <input
              name="email"
              onChange={[Function]}
              placeholder="example@gmail.com"
              required={true}
              type="email"
              value=""
            />
            <label
              forhtml="telephone"
            >
              <b>
                Phone number
              </b>
            </label>
            <input
              id="telephone"
              name="telephone"
              onChange={[Function]}
              pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
              placeholder="+256-703-492035"
              required={true}
              type="text"
              value=""
            />
            <label
              forhtml="password"
            >
              <b>
                Password
              </b>
            </label>
            <input
              id="pwd"
              name="password"
              onChange={[Function]}
              placeholder="***************"
              required={true}
              type="password"
              value=""
            />
            <input
              className="signupbtn"
              type="submit"
              value="Register"
            />
          </div>
        </form>
      </div>,
      "className": "flex-grow-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="modal-content animate"
          name="sigin"
          onSubmit={[Function]}
        >
          <h1>
            Fast-Food-Fast
          </h1>
          <div
            className="imgcontainer"
          />
          <div
            className="container"
          >
            <label
              forhtml="username"
            >
              <b>
                Username
              </b>
            </label>
            <input
              name="username"
              onChange={[Function]}
              placeholder="Enter Username"
              required={true}
              type="text"
              value=""
            />
            <label
              forhtml="Email"
            >
              <b>
                Email
              </b>
            </label>
            <input
              name="email"
              onChange={[Function]}
              placeholder="example@gmail.com"
              required={true}
              type="email"
              value=""
            />
            <label
              forhtml="telephone"
            >
              <b>
                Phone number
              </b>
            </label>
            <input
              id="telephone"
              name="telephone"
              onChange={[Function]}
              pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
              placeholder="+256-703-492035"
              required={true}
              type="text"
              value=""
            />
            <label
              forhtml="password"
            >
              <b>
                Password
              </b>
            </label>
            <input
              id="pwd"
              name="password"
              onChange={[Function]}
              placeholder="***************"
              required={true}
              type="password"
              value=""
            />
            <input
              className="signupbtn"
              type="submit"
              value="Register"
            />
          </div>
        </form>,
        "className": "signupmodal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Fast-Food-Fast
            </h1>,
            <div
              className="imgcontainer"
            />,
            <div
              className="container"
            >
              <label
                forhtml="username"
              >
                <b>
                  Username
                </b>
              </label>
              <input
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="Email"
              >
                <b>
                  Email
                </b>
              </label>
              <input
                name="email"
                onChange={[Function]}
                placeholder="example@gmail.com"
                required={true}
                type="email"
                value=""
              />
              <label
                forhtml="telephone"
              >
                <b>
                  Phone number
                </b>
              </label>
              <input
                id="telephone"
                name="telephone"
                onChange={[Function]}
                pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                placeholder="+256-703-492035"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="password"
              >
                <b>
                  Password
                </b>
              </label>
              <input
                id="pwd"
                name="password"
                onChange={[Function]}
                placeholder="***************"
                required={true}
                type="password"
                value=""
              />
              <input
                className="signupbtn"
                type="submit"
                value="Register"
              />
            </div>,
          ],
          "className": "modal-content animate",
          "name": "sigin",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fast-Food-Fast",
            },
            "ref": null,
            "rendered": "Fast-Food-Fast",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "imgcontainer",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  forhtml="username"
                >
                  <b>
                    Username
                  </b>
                </label>,
                <input
                  name="username"
                  onChange={[Function]}
                  placeholder="Enter Username"
                  required={true}
                  type="text"
                  value=""
                />,
                <label
                  forhtml="Email"
                >
                  <b>
                    Email
                  </b>
                </label>,
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="example@gmail.com"
                  required={true}
                  type="email"
                  value=""
                />,
                <label
                  forhtml="telephone"
                >
                  <b>
                    Phone number
                  </b>
                </label>,
                <input
                  id="telephone"
                  name="telephone"
                  onChange={[Function]}
                  pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                  placeholder="+256-703-492035"
                  required={true}
                  type="text"
                  value=""
                />,
                <label
                  forhtml="password"
                >
                  <b>
                    Password
                  </b>
                </label>,
                <input
                  id="pwd"
                  name="password"
                  onChange={[Function]}
                  placeholder="***************"
                  required={true}
                  type="password"
                  value=""
                />,
                <input
                  className="signupbtn"
                  type="submit"
                  value="Register"
                />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                    Username
                  </b>,
                  "forhtml": "username",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Username",
                  },
                  "ref": null,
                  "rendered": "Username",
                  "type": "b",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "username",
                  "onChange": [Function],
                  "placeholder": "Enter Username",
                  "required": true,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                    Email
                  </b>,
                  "forhtml": "Email",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email",
                  },
                  "ref": null,
                  "rendered": "Email",
                  "type": "b",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "example@gmail.com",
                  "required": true,
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                    Phone number
                  </b>,
                  "forhtml": "telephone",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Phone number",
                  },
                  "ref": null,
                  "rendered": "Phone number",
                  "type": "b",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "telephone",
                  "name": "telephone",
                  "onChange": [Function],
                  "pattern": "\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}",
                  "placeholder": "+256-703-492035",
                  "required": true,
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                    Password
                  </b>,
                  "forhtml": "password",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password",
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": "b",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "pwd",
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "***************",
                  "required": true,
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "signupbtn",
                  "type": "submit",
                  "value": "Register",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="signupmodal"
        >
          <form
            className="modal-content animate"
            name="sigin"
            onSubmit={[Function]}
          >
            <h1>
              Fast-Food-Fast
            </h1>
            <div
              className="imgcontainer"
            />
            <div
              className="container"
            >
              <label
                forhtml="username"
              >
                <b>
                  Username
                </b>
              </label>
              <input
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="Email"
              >
                <b>
                  Email
                </b>
              </label>
              <input
                name="email"
                onChange={[Function]}
                placeholder="example@gmail.com"
                required={true}
                type="email"
                value=""
              />
              <label
                forhtml="telephone"
              >
                <b>
                  Phone number
                </b>
              </label>
              <input
                id="telephone"
                name="telephone"
                onChange={[Function]}
                pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                placeholder="+256-703-492035"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="password"
              >
                <b>
                  Password
                </b>
              </label>
              <input
                id="pwd"
                name="password"
                onChange={[Function]}
                placeholder="***************"
                required={true}
                type="password"
                value=""
              />
              <input
                className="signupbtn"
                type="submit"
                value="Register"
              />
            </div>
          </form>
        </div>,
        "className": "flex-grow-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="modal-content animate"
            name="sigin"
            onSubmit={[Function]}
          >
            <h1>
              Fast-Food-Fast
            </h1>
            <div
              className="imgcontainer"
            />
            <div
              className="container"
            >
              <label
                forhtml="username"
              >
                <b>
                  Username
                </b>
              </label>
              <input
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="Email"
              >
                <b>
                  Email
                </b>
              </label>
              <input
                name="email"
                onChange={[Function]}
                placeholder="example@gmail.com"
                required={true}
                type="email"
                value=""
              />
              <label
                forhtml="telephone"
              >
                <b>
                  Phone number
                </b>
              </label>
              <input
                id="telephone"
                name="telephone"
                onChange={[Function]}
                pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                placeholder="+256-703-492035"
                required={true}
                type="text"
                value=""
              />
              <label
                forhtml="password"
              >
                <b>
                  Password
                </b>
              </label>
              <input
                id="pwd"
                name="password"
                onChange={[Function]}
                placeholder="***************"
                required={true}
                type="password"
                value=""
              />
              <input
                className="signupbtn"
                type="submit"
                value="Register"
              />
            </div>
          </form>,
          "className": "signupmodal",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Fast-Food-Fast
              </h1>,
              <div
                className="imgcontainer"
              />,
              <div
                className="container"
              >
                <label
                  forhtml="username"
                >
                  <b>
                    Username
                  </b>
                </label>
                <input
                  name="username"
                  onChange={[Function]}
                  placeholder="Enter Username"
                  required={true}
                  type="text"
                  value=""
                />
                <label
                  forhtml="Email"
                >
                  <b>
                    Email
                  </b>
                </label>
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="example@gmail.com"
                  required={true}
                  type="email"
                  value=""
                />
                <label
                  forhtml="telephone"
                >
                  <b>
                    Phone number
                  </b>
                </label>
                <input
                  id="telephone"
                  name="telephone"
                  onChange={[Function]}
                  pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                  placeholder="+256-703-492035"
                  required={true}
                  type="text"
                  value=""
                />
                <label
                  forhtml="password"
                >
                  <b>
                    Password
                  </b>
                </label>
                <input
                  id="pwd"
                  name="password"
                  onChange={[Function]}
                  placeholder="***************"
                  required={true}
                  type="password"
                  value=""
                />
                <input
                  className="signupbtn"
                  type="submit"
                  value="Register"
                />
              </div>,
            ],
            "className": "modal-content animate",
            "name": "sigin",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fast-Food-Fast",
              },
              "ref": null,
              "rendered": "Fast-Food-Fast",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "imgcontainer",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    forhtml="username"
                  >
                    <b>
                      Username
                    </b>
                  </label>,
                  <input
                    name="username"
                    onChange={[Function]}
                    placeholder="Enter Username"
                    required={true}
                    type="text"
                    value=""
                  />,
                  <label
                    forhtml="Email"
                  >
                    <b>
                      Email
                    </b>
                  </label>,
                  <input
                    name="email"
                    onChange={[Function]}
                    placeholder="example@gmail.com"
                    required={true}
                    type="email"
                    value=""
                  />,
                  <label
                    forhtml="telephone"
                  >
                    <b>
                      Phone number
                    </b>
                  </label>,
                  <input
                    id="telephone"
                    name="telephone"
                    onChange={[Function]}
                    pattern="\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}"
                    placeholder="+256-703-492035"
                    required={true}
                    type="text"
                    value=""
                  />,
                  <label
                    forhtml="password"
                  >
                    <b>
                      Password
                    </b>
                  </label>,
                  <input
                    id="pwd"
                    name="password"
                    onChange={[Function]}
                    placeholder="***************"
                    required={true}
                    type="password"
                    value=""
                  />,
                  <input
                    className="signupbtn"
                    type="submit"
                    value="Register"
                  />,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Username
                    </b>,
                    "forhtml": "username",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Username",
                    },
                    "ref": null,
                    "rendered": "Username",
                    "type": "b",
                  },
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "username",
                    "onChange": [Function],
                    "placeholder": "Enter Username",
                    "required": true,
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Email
                    </b>,
                    "forhtml": "Email",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email",
                    },
                    "ref": null,
                    "rendered": "Email",
                    "type": "b",
                  },
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "email",
                    "onChange": [Function],
                    "placeholder": "example@gmail.com",
                    "required": true,
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Phone number
                    </b>,
                    "forhtml": "telephone",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Phone number",
                    },
                    "ref": null,
                    "rendered": "Phone number",
                    "type": "b",
                  },
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "telephone",
                    "name": "telephone",
                    "onChange": [Function],
                    "pattern": "\\\\+[0-9]{1,3}-[0-9]{3}-[0-9]{6}",
                    "placeholder": "+256-703-492035",
                    "required": true,
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b>
                      Password
                    </b>,
                    "forhtml": "password",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Password",
                    },
                    "ref": null,
                    "rendered": "Password",
                    "type": "b",
                  },
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "id": "pwd",
                    "name": "password",
                    "onChange": [Function],
                    "placeholder": "***************",
                    "required": true,
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "signupbtn",
                    "type": "submit",
                    "value": "Register",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
